### PR TITLE
Email autocomplete for link shares

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Bring back email autocomplete for link shares.

## Related Issue
https://github.com/owncloud/core/issues/30670

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
I managed to recover this code from before the squash of https://github.com/owncloud/core/pull/30645.

It doesn't seem to work so maybe it wasn't the latest version. Or there were further changes.

@felixheidecke @DeepDiver1975

File: core/js/sharedialogmailview.js [hunk 1]
```diff
@@ -186,15 +186,37 @@
...
-					});
+					var data = [{
+						"id": query.term,
+						"text" : query.term
+					}];
+
+					// return query data ASAP
+					query.callback({results: data});
+
+					if (query.term.length >= 3) {
```
Comment (PVince81) [inline core/js/sharedialogmailview.js]: plug `OC.getCapabilities().files_sharing.search_min_length` here.

**Comment by DeepDiver1975** [comment]
What is this? @PVince81 @tomneedham 

```
There was 1 failure:

1) Test\Repair\RepairOrphanedSubshareTest::testLargeSharesWithOrphans
Failed asserting that 82 matches expected 81.

/drone/src/tests/lib/Repair/RepairOrphanedSubshareTest.php:288

--

**Comment by PVince81** [comment]
- [ ] add search_min_length on backend as well

**Comment by PVince81** [comment]
Tested:
- [x] TEST: local users not returned
- [x] TEST: federated users not returned
- [ ] TEST: contacts from contacts app returned

- [ ] TEST: min-length respected on client side
- [ ] TEST: min-length respected on backend :no_entry_sign:

**Comment by PVince81** [comment]
:+1: for the changes from @felixheidecke @DeepDiver1975  (I can't approve my own PR)

CI will likely fail randomly due to issues, waiting for https://github.com/owncloud/core/pull/30992

**Comment by PVince81** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by PVince81** [comment]
stable10: https://github.com/owncloud/core/pull/30998

**Pull request merged**